### PR TITLE
Fix error messages, add loss functions

### PR DESCRIPTION
**Pull request by aaronhma**
### Summary
This PR is fixing some of Keras error messages and also re-writing and adding some loss functions.
#### Re-write Loss functions:
* mean_squared_error (faster)
* mean_absolute_error (faster)

#### Added new Loss functions:
* root_mean_square_error (new)
* rms_log_error (new)
* mean_bias_deviation (new)

### Related Issues
* I have tested it on Mac, Linux, and Windows, and it seems to work fine.

### PR Overview
- [ X ] This PR requires new unit tests [y/n] N
- [ X ] This PR requires to update the documentation [y/n] N
- [ X ] This PR is backward compatible [y/n] Y
- [ X ] This PR changes the current API [y/n] N

### CODE REVIEW

File: keras/backend/common.py [hunk 1]
```diff
@@ -80,7 +80,8 @@ def set_floatx(floatx):
     """
     global _FLOATX
     if floatx not in {'float16', 'float32', 'float64'}:
-        raise ValueError('Unknown floatx type: ' + str(floatx))
+        raise ValueError('Unknown float datatype: ' + str(floatx))
+        raise ValueError('Only these data types are supported: float16, float32 and float64. Thanks! 😀')
```
Comment (fchollet) [inline keras/backend/common.py]: It is a good idea to list the supported types. A good error message should also print what was passed (like in the previous message). Also a good error message is informative and to the point, so no need for `Thanks! 😀` even though I do appreciate the positivity here :)

File: keras/callbacks.py [hunk 2]
```diff
@@ -26,6 +26,7 @@
     import requests
 except ImportError:
     requests = None
+    print("Failed to import requests.")
```
Comment (fchollet) [inline keras/callbacks.py]: This is not necessary. It's ok not to have requests installed (until you try to use the callback that uses it).

File: keras/callbacks.py [hunk 3]
```diff
@@ -1019,6 +1020,7 @@ def __init__(self, log_dir='./logs',
         except ImportError:
             raise ImportError('You need the TensorFlow module installed to '
                               'use TensorBoard.')
+            raise ImportError('Don\'t give up! Just do: pip3 install tensorflow 😀')
```
Comment (fchollet) [inline keras/callbacks.py]: You can just append:
```
To install TensorFlow: `pip3 install tensorflow` 
```
to the previous message.

File: keras/constraints.py [hunk 4]
```diff
@@ -182,5 +182,5 @@ def get(identifier):
     elif callable(identifier):
         return identifier
     else:
-        raise ValueError('Could not interpret constraint identifier: ' +
+        raise ValueError('Failed to interpret constraint identifier: ' +
```
Comment (fchollet) [inline keras/constraints.py]: Changing this breaks the consistency of these error messages across submodules, please revert.

File: keras/losses.py [hunk 5]
```diff
@@ -8,14 +8,19 @@
 from . import backend as K
 from .utils.generic_utils import deserialize_keras_object
 from .utils.generic_utils import serialize_keras_object
+import numpy as np
 
 
 def mean_squared_error(y_true, y_pred):
-    return K.mean(K.square(y_pred - y_true), axis=-1)
-
+    return np.square(np.subtract(y_pred,y_true)).mean()
```
Comment (fchollet) [inline keras/losses.py]: 2 things here:
- these functions are meant to return arrays of samplewise loss scores, not a single scalar. That's what we have the `axis=-1` argument.
- these functions are meant to operate on symbolic tensors, not Numpy arrays. That's why we use `K` methods instead of `np`.

File: keras/losses.py [hunk 6]
```diff
@@ -92,16 +97,58 @@ def cosine_proximity(y_true, y_pred):
...
+def mean_bias_deviation(y_pred, y_true):
+    """
+    Mean Bias Deviation
+    """
+    predict = np.array(y_pred)
+    actual = np.array(y_true)
+
+    difference = predict - actual
+    numerator = np.sum(difference) / len(predict) 
+    denumerator =  np.sum(actual) / len(predict)
```
Comment (fchollet) [inline keras/losses.py]: It's "denominator"

File: keras/losses.py [hunk 7]
```diff
@@ -92,16 +97,58 @@ def cosine_proximity(y_true, y_pred):
     y_pred = K.l2_normalize(y_pred, axis=-1)
     return -K.sum(y_true * y_pred, axis=-1)
 
+def root_mean_square_error(y_pred, targets):
```
Comment (fchollet) [inline keras/losses.py]: Not that square_root(a + b) != square_root(a) + square_root(b)

therefore square root cannot be used as a loss function, since we reduce per-batch losses via addition and subtraction.

This is also true for rms_log_error.
